x/valset-pref/keeper_test.go

```diff
@@ -306,67 +307,97 @@ func (s *KeeperTestSuite) TestGetValSetPreferencesWithDelegations() {
 func (s *KeeperTestSuite) TestFormatToValPrefArr() {
 	s.SetupTest()
 
-	// prepare existing delegations validators
-	valAddrs := s.SetupMultipleValidators(3)
+	tests := map[string]struct {
+		delegationShares       []osmomath.Dec
+		expectedValPrefWeights []osmomath.Dec
+		invalidDelegation      bool
 
-	tests := []struct {
-		name             string
-		delegations      []stakingtypes.Delegation
-		expectedValPrefs []types.ValidatorPreference
+		expectedError error
 	}{
-		{
-			name: "Single delegation",
-			delegations: []stakingtypes.Delegation{
-				{
-					ValidatorAddress: valAddrs[0],
-					Shares:           sdk.NewDec(100),
-				},
-			},
-			expectedValPrefs: []types.ValidatorPreference{
-				{
-					ValOperAddress: valAddrs[0],
-					Weight:         sdk.NewDec(1),
-				},
-			},
+		"Single delegation": {
+			delegationShares:       []osmomath.Dec{osmomath.NewDec(100)},
+			expectedValPrefWeights: []osmomath.Dec{osmomath.NewDec(1)},
```

Comment:
Do you know if we are able to make a test case that has a certain amount of shares that corresponds to a non Int number of underlying tokens? This way we can test the rounding behavior?

My guess is this would involving slashing one validator after the initial delegation. We could even do it for every test iteration and not just special case it. Lmk what you think